packages/featureflags/featureToggleManager.js

```diff
@@ -7,89 +7,101 @@ const { getRedisClient } = require('@open-condo/keystone/redis')
         }
+        this._redisKey = REDIS_FEATURES_KEY
+        this._redisExpires = FEATURES_EXPIRED_IN_SECONDS
     }
 
     async fetchFeatures () {
-        try {
-            const redisClient = getRedisClient()
-            const cachedFeatureFlags = await redisClient.get(REDIS_FEATURES_KEY)
-
-            if (cachedFeatureFlags) return JSON.parse(cachedFeatureFlags)
+        if (this._url) {
+            try {
+                const cachedFeatureFlags = await this.redis.get(this._redisKey)
+                if (cachedFeatureFlags) return JSON.parse(cachedFeatureFlags)
 
-            if (featureToggleApiUrl && featureToggleApiKey) {
-                const fetchedFeatureFlags = await fetch(`${featureToggleApiUrl}/${featureToggleApiKey}`)
+                const fetchedFeatureFlags = await fetch(this._url)
                     .then((res) => res.json())
                     .then((parsed) => {
                         return Promise.resolve(parsed.features)
                     })
 
-                redisClient.set(REDIS_FEATURES_KEY, JSON.stringify(fetchedFeatureFlags), 'EX', FEATURES_EXPIRED_IN_SECONDS)
-
+                await this.redis.set(this._redisKey, JSON.stringify(fetchedFeatureFlags), 'EX', this._redisExpires)
                 return fetchedFeatureFlags
-            } else {
-                throw new Error(WRONG_FEATURE_TOGGLE_CONFIG_ERROR)
+            } catch (err) {
+                logger.error({ msg: 'fetchFeatures error', err })
             }
-        } catch (e) {
-            logger.error({ msg: 'fetchFeatures error', error: e })
-
-            return {}
+        } else if (this._static) {
+            return JSON.parse(JSON.stringify(this._static))
         }
-    }
 
-    async #getContext (keystoneContext) {
-        return get(keystoneContext, 'req', null)
-            ? keystoneContext
-            : { req: { features: await this.fetchFeatures() } }
+        throw new Error('FeatureToggleManager config error!')
     }
 
-    async isFeatureEnabled (keystoneContext, featureName, featuresContext) {
-        const context = await this.#getContext(keystoneContext)
-        const request = context.req
-        const headersFeatureFlags = get(request, ['headers', 'feature-flags'])
-
-        // Here it will stop under tests
-        if (conf.NODE_ENV === 'test') return headersFeatureFlags === 'true'
+    async #getFeaturesFromKeystoneContext (keystoneContext) {
+        const req = get(keystoneContext, 'req')
+        let features = get(req, 'features')
 
-        const growthbook = new GrowthBook()
+        // Note: fetch features if needed! And save it in `req` if in request context
+        if (!features) {
+            features = await this.fetchFeatures()
+            if (req) req.features = features
+        }
 
-        growthbook.setFeatures(request.features)
+        return features
+    }
 
+    async #getGrowthBookInstance (keystoneContext, featuresContext) {
```

Comment:
Why do you use `#method` for methods and `_field` for private fields mixing 2 code styles into 1?

Reply:
We need to standartize it %) 
Need to fix it!